src/os/bluestore/BlueStore.cc

```diff
@@ -6282,6 +6282,7 @@ void BlueStore::_txc_state_proc(TransContext *txc)
     case TransContext::STATE_PREPARE:
       txc->log_state_latency(logger, l_bluestore_state_prepare_lat);
       if (txc->ioc.has_pending_aios()) {
+	txc->is_pipelined_io = true;
 	txc->state = TransContext::STATE_AIO_WAIT;
 	_txc_aio_submit(txc);
 	return;
```

Comment:
..I think were we need an else case where we look for txc's ahead of us that are doing io.  I think it's sufficient to check that if hte next txc is >= KV_DONE then we're safe, otherwise set is_pipelined_io?

Alternatively, flip this bool around to be safe_for_sync_commit or something...